lambda/custom/state_handlers.js

```diff
@@ -521,21 +612,36 @@ function synthesisHelperUrl(stateObj) {
       }
     );
     console.log('Chosen Article is: ' + stateObj.attributes['chosenArticle']);
-    const getArticle = scout_agent
-      .handleUrl(stateObj.attributes['chosenArticle'], stateObj.event)
-      .then(
-        url => {
-          console.log('promise resolved: ' + url.url);
-          stateObj.attributes['url'] = url.url;
-          stateObj.attributes['offsetInMilliseconds'] = 0;
-          audio_controller.play.call(stateObj);
-        },
-        error => {
-          console.log('handleURL promise failed');
-          stateObj.response.speak(constants.strings.ARTICLE_FAIL_MSG);
-          stateObj.emit(':responseReady');
-        }
+    const getArticle = scout_agent.handleUrl(
+      stateObj.attributes['chosenArticle'],
+      stateObj.event
+    );
+    stateObj.attributes['full'] =
+      stateObj.event.request.intent.name == 'fullarticle';
+    stateObj.attributes['userId'] = stateObj.event.session.user.accessToken;
+    let getOffset;
+    if (stateObj.attributes['full']) {
+      getOffset = scout_agent.getArticleStatus(
+        stateObj.event.session.user.accessToken,
+        stateObj.attributes['articleId']
       );
+    }
+
+    Promise.all([getArticle, getOffset])
```

Comment:
These variable names are a bit confusing -- especially in JS where functions can get passed around in parameters like this, I'd stay away from including verbs in your object names. Just going with `article` and `offset` would read better.